json/src/main/scala/fs2/data/json/package.scala

```diff
@@ -143,6 +143,43 @@ package object json {
 
   }
 
+  object unwrap {
+
+    /** Strips the top-level array from a stream of Json tokens.
+      *
+      * If the stream does not start with `[` and end with `]`, an error is raised.
+      */
+    def stripTopLevelArray[F[_]: RaiseThrowable]: Pipe[F, Token, Token] = { tokens =>
+      def invalidFirst(first: Option[Token]): JsonException =
+        JsonException(s"Expected start of array, got: $first")
+
+      def invalidLast(last: Option[Token]): JsonException =
+        JsonException(s"Expected end of array, got: $last")
+
+      def init(stream: Stream[F, Token]): Pull[F, Token, Unit] =
+        stream.pull.uncons.flatMap {
+          case Some((chunk, tail)) =>
+            if (chunk.head.forall(_ == Token.StartArray))
+              loop(tail, chunk.drop(1))
+            else
+              Pull.raiseError(invalidFirst(chunk.head))
+          case None => Pull.done
```

Comment:
For consistency, don't we want to fail in the empty case as well if we are strict?

Reply:
Done